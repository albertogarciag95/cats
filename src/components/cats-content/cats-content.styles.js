import styled from 'styled-components';

export const CatCardsContainer = styled.div`
  margin: 20px;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  overflow-y: auto;
`;

export const ShuffleButton = styled.button`
  font-size: 2em;
  border-radius: 30px;
  background-color: orange;
  font-family: 'Chango', cursive;
  margin: 1em;
  cursor: pointer;
  min-width: 200px;
`;

export const CatsContentWrapper = styled.div`
  width: 60%;
  display: inline-flex;
  flex-direction: column;
  justify-content: space-between;
  margin: 0 auto;
  height: 90%;
`;
