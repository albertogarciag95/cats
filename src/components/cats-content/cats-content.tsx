import React, { useEffect, useState } from 'react';
import { CatCard } from '../cat-card/cat-card';
import { 
  CatCardsContainer, 
  ShuffleButton, 
  CatsContentWrapper
} from './cats-content.styles';

export const CatsContent = () => {
  const [catFacts, setCatFacts] = useState([]);
  
  useEffect(() => {
    fetchCatsData()
  }, []);

  const fetchCatsData = () => {
    return fetch('https://cat-fact.herokuapp.com/facts/random?animal_type=cat&amount=4')
      .then(response => response.json())
      .then(responseCatFacts => setCatFacts(responseCatFacts))
  }

  return (
    <CatsContentWrapper>
      <CatCardsContainer>
        {catFacts.map(({ text }) => (
          <CatCard description={text} key={text} />
        ))}
      </CatCardsContainer>
      <ShuffleButton onClick={fetchCatsData}>Shuffle</ShuffleButton>
    </CatsContentWrapper>
  );
};