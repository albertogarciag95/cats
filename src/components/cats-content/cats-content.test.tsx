import renderer, { act } from 'react-test-renderer';
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from 'enzyme';

import { CatsContent } from './cats-content';

configure({ adapter: new Adapter() });


test('CatsContent match snapshot', () => {
  const component = renderer.create(
    <CatsContent />
  );
  let cardlink = component.toJSON();
  expect(cardlink).toMatchSnapshot();
});

test('Shuffle button fires fetch data', async () => {
  const component = shallow(<CatsContent />)
  await act(async () => {
    await component.find('ShuffleButton').props().onClick();
  })
  expect(component.find('CatCard')).toHaveLength(4);
});
