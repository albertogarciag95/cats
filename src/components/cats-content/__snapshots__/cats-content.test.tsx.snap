// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatsContent match snapshot 1`] = `
<div
  className="sc-jrsJWt hvEvPe"
>
  <div
    className="sc-fujyAs JERqq"
  />
  <button
    className="sc-pNWdM jQTDcs"
    onClick={[Function]}
  >
    Shuffle
  </button>
</div>
`;
